src/server/models/cacheable_queries/user.js

```diff
@@ -1,79 +1,145 @@
+const userAuthKey = (authId) => `${process.env.CACHE_PREFIX || ''}texterauth-${authId}`
+
+const getHighestRolesPerOrg = (userOrgs) => {
+  const highestRolesPerOrg = {}
+  userOrgs.forEach(userOrg => {
+    const orgId = userOrg.organization_id
+    const orgRole = userOrg.role
+    const orgName = userOrg.name
+
+    if (highestRolesPerOrg[orgId]) {
+      if (isRoleGreater(
+        orgRole, highestRolesPerOrg[orgId].role
+      )) {
+        highestRolesPerOrg[orgId].role = orgRole
       }
-      highestRole = getHighestRole(userRoles.map((r) => r.role))
-      await r.redis.hsetAsync(userKey, orgId, highestRole)
+    } else {
+      highestRolesPerOrg[orgId] = { id: orgId, role: orgRole, name: orgName }
     }
-    return (acceptableRoles.indexOf(highestRole) >= 0)
-  } else {
-    // regular DB approach
-    const userHasRole = await r.getCount(
-      r.knex('user_organization')
-        .where({ user_id: userId,
-                 organization_id: orgId })
-        .whereIn('role', acceptableRoles)
-    )
-    return userHasRole
-  }
+  })
+  return highestRolesPerOrg
 }
 
-export async function userLoggedIn(authId) {
-  const authKey = `texterauth-${authId}`
+const dbLoadUserRoles = async (userId) => {
+  const userOrgs = await r.knex('user_organization')
+    .where('user_id', userId)
+    .join('organization', 'user_organization.organization_id', 'organization.id')
+    .select('user_organization.role', 'user_organization.organization_id', 'organization.name')
+
+  const highestRolesPerOrg = getHighestRolesPerOrg(userOrgs)
 
   if (r.redis) {
-    const cachedAuth = await r.redis.getAsync(authKey)
-    if (cachedAuth) {
-      return JSON.parse(cachedAuth)
-    }
+    // delete keys first
+    // pass all values to hset instead of looping
+    const mappedHighestRoles = Object.values(highestRolesPerOrg).reduce((acc, orgRole) => {
+      acc.push(orgRole.id, `${orgRole.role}:${orgRole.name}`)
+      return acc
+    }, [])
+
+    await r.redis.multi()
+      .del(userRoleKey(userId))
+      .hmset(userRoleKey(userId), ...mappedHighestRoles)
+      .execAsync()
   }
 
+  return highestRolesPerOrg
+}
+
+const dbLoadUserAuth = async (authId) => {
   const userAuth = await r.knex('user')
     .where('auth0_id', authId)
     .select('*')
     .first()
 
   if (r.redis && userAuth) {
+    const authKey = userAuthKey(authId)
     await r.redis.multi()
       .set(authKey, JSON.stringify(userAuth))
       .expire(authKey, 86400)
       .execAsync()
+    await dbLoadUserRoles(userAuth.id)
   }
   return userAuth
 }
 
-export async function currentEditors(redis, campaign, user) {
-  // Add user ID in case of duplicate admin names
-  const displayName = `${user.id}~${user.first_name} ${user.last_name}`
+const userHasRole = async (userId, orgId, acceptableRoles) => {
+  if (r.redis) {
+    // cached approach
+    const userKey = `texterinfo-${userId}`
+    let highestRole = await r.redis.hgetAsync(userKey, orgId)
```

Comment:
nit: make this something like `const userRoleResult` or something because it's *not* the highestRole